lib/components/fetch_data/data_list.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:provider/provider.dart';
+import 'package:subscription_manager/components/fetch_data/subscribe_item.dart';
+import 'package:subscription_manager/data/data_model.dart';
+
+class DataList extends StatelessWidget {
+  const DataList({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final data = Provider.of<List<SubscriptionData>>(context);
```

Comment:
Your bloc usage here is completely wrong. You use provider to get a bloc, and then you get data from bloc (and data is stream). Here, each of your elements is bloc by itself (so instead of creating a bloc - container - that has data, you are making your bloc an actual piece of data), and you are resolving the list of blocs,